### ISSUE TITLE
Streamline Slab types

### ISSUE THREAD

**Issue by kmdeck** [CliMA/ClimaCoupler.jl#59]
Currently, slab models are used for land (in some cases), sea ice, and ocean. Land can also be a bucket model, and ocean temperatures can also be prescribed. In the future, there will be different sea ice models as well.
- use multiple dispatch for obtaining T, energy, etc correctly from different models, as they are stored differently in different models
- make a more useful slab_utils file and separate ocean, land, and ice directories?

Basically, clean up duplicate code between slab models and make use of dispatch over flags for different model types.

**Comment by LenkaNovak** [comment]
To be addressed as part of #212
